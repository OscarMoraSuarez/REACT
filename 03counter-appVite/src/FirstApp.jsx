import PropTypes from 'prop-types';


/* const saludar=(nombre)=>{

    return nombre.toUpperCase();

} */

const FirstApp = ({title,subtitle,name }) => {



            return (
                    <>      
                        <h1 data-testid="test-title">{title}</h1> 
                        <p>{subtitle}</p>
                        <p>{subtitle}</p>
                        <p>{subtitle}</p>
                        <span>{name}</span>
                    </>
                )
}

export{

    FirstApp

}

FirstApp.propTypes={
    title:PropTypes.string.isRequired,
    subtitle: PropTypes.string,



}

/* estas propiedades entran antes que lso PropTypes */
FirstApp.defaultProps={
    name:'RaJSo',
    subtitle:'no hay subtitulo',
    /* title:'No hay titulo', */

     

}