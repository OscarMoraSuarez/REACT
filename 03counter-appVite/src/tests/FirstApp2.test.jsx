import { render,screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp"

describe('Pruebas en <FirsApp/>', () => { 
    const title="Hola, soy Goku";
    const subtitle='Soy un subtitulo';
    test('debe hacer match con el sanpshot', () => { 

        const {container} =render(<FirstApp title={title}/>);
        
        expect(container).toMatchSnapshot();


    });

    test('deb mostrar el mensaje "Hola, soy Goku"', () => { 
        
        render(<FirstApp title={title}/>)
        /* screen.debug()  despues de el render muestra el body del componente*/
        // expect(screen.getByText(title)).not.toBeTruthy();
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Deb mostrar el titulo n un h1', () => { 

        render(<FirstApp title={title} />);
        /* utilizo el es scrren para acceder a los roles que son elementos html en este caso un encabezado
        nivel 1 osea ''h1 */
        expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title)


     })
     test('debe mostrar el subtitulo enviado por props', () => { 

         render(<FirstApp 
            title={title} 
            subtitle={subtitle}
            />);

         expect(screen.getAllByText(subtitle).length).toBe(3);    
      })

})