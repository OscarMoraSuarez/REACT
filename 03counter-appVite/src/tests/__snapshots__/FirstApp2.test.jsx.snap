// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirsApp/> debe hacer macth con el sanpshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay subtitulo
  </p>
  <span>
    RaJSo
  </span>
</div>
`;
